exercises/practice/mazy-mice/src/test/java/MazeGeneratorTest.java

```diff
@@ -0,0 +1,259 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Ignore;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+public class MazeGeneratorTest {
+    private static final char EMPTY_CELL = ' ';
+    private static final Set<Character> ALLOWED_SYMBOLS = Set.of(
+            EMPTY_CELL, // space
+            '┌', // box drawings light down and right
+            '─', // box drawings light horizontal
+            '┬', // box drawings light down and horizontal
+            '┐', // box drawings light down and left
+            '│', // box drawings light vertical
+            '└', // box drawings light up and right
+            '┴', // box drawings light up and horizontal
+            '┘', // box drawings light up and left
+            '├', // box drawings light vertical and right
+            '┤', // box drawings light vertical and left
+            '┼', // box drawings light vertical and horizontal
+            '⇨'  // rightwards white arrow
+    );
+    private static final char VISITED_CELL = '.';
+    private static final int RECTANGLE_ROWS = 6;
+    private static final int RECTANGLE_COLUMNS = 18;
+    private static final int SEED_ONE = 42;
+    private static final int SEED_TWO = 43;
+    private MazeGenerator sut;
+
+    @Before
+    public void setup() {
+        sut = new MazeGenerator();
+    }
+
+    @Test
+    public void mazeIsNotNull() {
+        var maze = sut.generatePerfectMaze(RECTANGLE_ROWS, RECTANGLE_COLUMNS);
+
+        assertThat(maze)
+                .as("The maze is not null.")
+                .isNotNull();
+
+        Arrays.stream(maze)
+                .map(String::new)
+                .forEach(System.out::println);
+    }
```

Comment:
This test is not listed in the canonical data, and I don't think it makes a lot of sense to add it honestly. Students are never instructed to return `null` from an exercise, so if they decide to do so IMO it would be fine if the other tests all fail on a `NullPointerException`.

Reply:
deleted